### PR TITLE
Add new item form

### PR DESCRIPTION
**Pull request by Middlebrooks314**
### Summary
The `post-item-page` branch is adding the user functionality for adding new items to the store. When the user goes to the ~New Item", they can add a new item. After submitting the new item, they are taken to the item show page for the newly added item. 

https://user-images.githubusercontent.com/43938783/118580189-135e9180-b755-11eb-9e4d-d8927e239251.mov

### CODE REVIEW

File: src/apiHandler.js [hunk 1]
```diff
@@ -19,17 +19,37 @@ class ApiHandler {
...
-      headers: { accept: "application/json" }
+      headers: { accept: 'application/json' }
     })
     return data
   }
+
+  handleSubmit(event, formData, history){
+    event.preventDefault()
+    this.postNewItem(formData, history)
+    console.log('submitted')
```
Comment (cmvandrevala) [inline src/apiHandler.js]: Request for Change - Get rid of `console.log`

File: src/apiHandler.js [hunk 2]
```diff
@@ -19,17 +19,37 @@ class ApiHandler {
...
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(formData)
+  };
+    await fetch(`${this.url}/items`, requestOptions)
+        .then(response => response.json())
+        .then((item) => {
+          history.replace(`/items/${item.id}`)
+        })
+        .catch(error => console.log(error))
```
Comment (cmvandrevala) [inline src/apiHandler.js]: Request for Change - Get rid of `console.log`

File: src/components/itemsIndex/itemsIndex.js [hunk 3]
```diff
@@ -17,9 +17,9 @@ const ItemsIndex = (props) => {
       </tr>
     </thead>
     <tbody>
-      {items.map((item, index) => (
+      {items.map((item) => (
         <tr key={item.id}>
-          <td data-testid="item-name"><Link to={`/items/${index + 1}`}>{item.name}</Link></td>
+          <td data-testid="item-name"><Link to={`/items/${item.id}`}>{item.name}</Link></td>
```
Comment (cmvandrevala) [inline src/components/itemsIndex/itemsIndex.js:22]: Praise - Good change

**Review by cmvandrevala — APPROVED** [review]

**Review by mbdebbeler — APPROVED** [review]
Looks good to me too! Take out those `console.log()`s and ship it.

**Pull request merged**